1 121 define a number of key cities and stations around switzerland and france

fkapsahili opened the pull request:
This pull request closes Issue 1.2.1 define a number of key cities and stations around Switzerland and France #1 as part of Iteration 1.

In the `README.md` of the `script` folder I have listed all prerequisites that must be given to be able to execute the script.

cgurtner commented on script/find_cities_and_stations.py:
```diff
@@ -0,0 +1,228 @@
+                    "username": username,
+                },
+            )
+            return near_by_cities.json()["geonames"]
```
This throws an error for me. If I `print(near_by_cities.json())` I get the response:

`{'status': {'message': 'missing parameter ', 'value': 14}}`

cgurtner commented on script/README.md:
```diff
@@ -0,0 +1,36 @@
+
+- requests
+- requests_cache
+- dotenv
```
Isn't it called `python-dotenv`?

refs https://pypi.org/project/python-dotenv/

fkapsahili commented:
@cgurtner I also got the error when the google api key was not created in the .env file in the root folder. With the improved error handling in [92fea03](https://github.com/cgurtner/PROG2-P05/pull/6/commits/92fea032af8aacf58eb3735113fea385e263d754), this should now also be output on the console.

cgurtner commented:
Ok, works now. But I'm not sure if it is correct regarding the assignment. 

referencing https://github.com/cgurtner/PROG2-P05/issues/1#issuecomment-1113958194

How do I fetch the stations around the border? If I enter an address, it fetches stations nearby, but not around the border.

fkapsahili commented:
Currently there is no way or request parameter to query the borders between the two countries via the Google Maps API:
- https://developers.google.com/maps/documentation/places/web-service/supported_types
- https://stackoverflow.com/questions/11207045/google-has-started-highlighting-search-areas-in-pink-color-is-this-feature-avai

Basically, my implementation searches for the provided address (or the default address near the borders if not specified) and then tries to find cities and stations nearby.

I have now added a default value to my implementation, so that without specifying a parameter when running the script, stations and cities near the borders of Switzerland and France will be found successfully:
```bash
python find_cities_and_stations.py
```
Also note that I have added the pycountry package.
If you have an idea how we could improve it, let me know.

cgurtner commented:
It seems like I'm still not getting it or I'm using the script wrong. If I execute the script I receive stations in France and Switzerland. But we should habe stations around the borders of France and Switzerland. 

My idea would be: Place a address near the border of France and Switzerland for those marked blue in the assignment. Search for stations for each country nearby but filter them to NOT be in France or Switzerland. 

Wouldn't that do the trick? Sry if I'm not understanding your implementation correctly.

fkapsahili commented:
Thanks for your explanations, I must have misunderstood something. Should now work after the current commit, so that only cities and stations outside the two countries (CH, FR) or excluded countries are returned. I have also documented this in the readme.

cgurtner reviewed (approved):
looks good to me, thanks
